WalletWasabi.Gui/Tabs/LegalDocumentsViewModel.cs

```diff
@@ -1,15 +1,58 @@
 using System;
 using System.IO;
+using System.Reactive;
+using System.Reactive.Linq;
 using Avalonia;
+using Avalonia.Logging;
 using Avalonia.Platform;
+using AvalonStudio.Extensibility;
+using AvalonStudio.Shell;
+using ReactiveUI;
+using Splat;
+using WalletWasabi.Gui.Helpers;
+using WalletWasabi.Gui.Models.TextResourcing;
 using WalletWasabi.Gui.ViewModels;
+using WalletWasabi.Legal;
 
 namespace WalletWasabi.Gui.Tabs
 {
 	public class LegalDocumentsViewModel : TextResourceViewModelBase
 	{
-		public LegalDocumentsViewModel() : base("Legal Documents", new Uri("avares://WalletWasabi.Gui/Assets/LegalDocuments.txt"))
+		public ReactiveCommand<Unit, Unit> AgreeClicked { get; set; }
+		public LegalDocuments LegalDoc { get; }
+
+		public bool IsAgreed { get; set; }
+
+		public LegalDocumentsViewModel(string content = null, LegalDocuments legalDoc = null) : base("Legal Documents", new TextResource() { FilePath = legalDoc?.FilePath, Content = content })
+		{
+			LegalDoc = legalDoc;
+			IsAgreed = content is null; // If content wasn't provided, then the filepath must had been provided. If the file exists, then it's agreed.
+
+			AgreeClicked = ReactiveCommand.CreateFromTask(async () =>
+			{
+				IsAgreed = true;
+				await LegalDoc.ToFileAsync(TextResource.Content);
```

Comment:
Move business logic into LegalDoc. Create method something like called SetAgree(). So later if we need to add more code we have to tinker in LegalDocuments class.

Reply:
I don't know how to do this. You can think about it this way: `ToFileAsync` is the `SetAgree`.

Reply:
> If it seems not possible to implement something without breaking some of this advice please consult with @danwalmsley.

It seems to me it's not possible to implement it without "Putting buisness logic inside control code".  

Do you have an idea?

Reply:
This is good as it is. We cannot move this until we have something like a LegalDocService which we agreed to not have. See my comment here: https://github.com/zkSNACKs/WalletWasabi/pull/2805#issuecomment-577590661